roles/kubernetes/preinstall/tasks/0120-growpart-azure-centos-7.yml

```
@@ -7,25 +7,34 @@
     name: cloud-utils-growpart
     state: present
 
+- name: Search root filesystem device
+  vars:
+    query: "[?mount=='/'].device"
+    _root_device: "{{ ansible_mounts|json_query(query) }}"
+  set_fact:
+    device: "{{ _root_device | first | regex_replace('([^0-9]+)[0-9]+', '\\1') }}"
+    partition: "{{ _root_device | first | regex_replace('[^0-9]+([0-9]+)', '\\1') }}"
+    root_device: "{{ _root_device }}"
```

I tested this task locally with `/dev/sda1` as `_root_device`.
`device` and `partition` would be
```
    "ansible_facts": {
        "device": "/",
        "partition": "/",
        "root_device": "/dev/sda1"
    },
```
I think that is different from the expected.

---

```suggestion
    device: "{{ _root_device | regex_replace('([^0-9]+)[0-9]+', '\\1') }}"
    partition: "{{ _root_device | regex_replace('[^0-9]+([0-9]+)', '\\1') }}"
```
ah, if removing `| first ` from the above 2 parts, that would work fine:
```
    "ansible_facts": {
        "device": "/dev/sda",
        "partition": "1",
        "root_device": "/dev/sda1"
    },
```

---

I'm guessing you set `_root_device` as a string, but it should be a list because ansible returns an array of results, one per node, please, try running this playbook to test it:
```
---
- hosts: localhost
  tasks:
    - name: facts
      vars:
        query: "[?mount=='/'].device"
        _root_device: "{{ ansible_mounts|json_query(query) }}"
      set_fact:
        device: "{{ _root_device | first | regex_replace('([^0-9]+)[0-9]+', '\\1') }}"
        partition: "{{ _root_device | first | regex_replace('[^0-9]+([0-9]+)', '\\1') }}"
        root_device: "{{ _root_device }}"

    - name: debug
      debug:
        var: "{{ item }} "
      loop:
        - root_device
        - device
        - partition

```

---

Thanks for your explanation, I can see the point.

/lgtm